code/modules/reagents/chemistry/machinery/reagentgrinder.dm

```diff
@@ -64,334 +48,493 @@
+		if(weapon.w_class + total_weight > maximum_weight)
+			to_chat(user, span_warning("[weapon] is too big to fit into [src]"))
+			continue
+		weapon.forceMove(src)
+		total_weight += weapon.w_class
+		items_transfered += 1
+		to_chat(user, span_notice("[weapon] was loaded into [src]"))
+
+	return items_transfered
+
+/obj/machinery/reagentgrinder/item_interaction(mob/living/user, obj/item/tool, list/modifiers, is_right_clicking)
+	if(user.combat_mode || (tool.item_flags & ABSTRACT) || (tool.flags_1 & HOLOGRAM_1) || !can_interact(user) || !user.can_perform_action(src, ALLOW_SILICON_REACH))
+		return ..()
+
+	//add the beaker
+	if (istype(tool, /obj/item/reagent_containers/cup/beaker))
+		replace_beaker(user, tool)
+		to_chat(user, span_notice("You add [tool] to [src]."))
+		return ITEM_INTERACT_SUCCESS
+
+	//add items from bag
+	else if(istype(tool, /obj/item/storage/bag))
+		var/list/obj/item/to_add = list()
+		//list of acceptable items from the bag
+		var/static/list/accepted_items = list(
+			/obj/item/grown,
+			/obj/item/food/grown,
+			/obj/item/food/honeycomb,
+		)
+
+		//add to list of items to check for
+		for(var/obj/item/ingredient in tool)
+			if(!is_type_in_list(ingredient, accepted_items))
+				continue
+			to_add += ingredient
+
+		//add the items
+		var/items_added = load_items(user, to_add)
+		if(!items_added)
+			to_chat(user, span_warning("No items were added"))
+			return ITEM_INTERACT_BLOCKING
+		to_chat(user, span_notice("[items_added] items were added from [tool] to [src]"))
+		return ITEM_INTERACT_SUCCESS
+
+	//add item directly
+	else if(tool.grind_results || tool.juice_typepath)
+		if(tool.atom_storage) //anything that has internal storage would be too much recursion for us to handle
+			to_chat(user, span_notice("Drag this item onto [src] to dump its contents"))
+			return ITEM_INTERACT_BLOCKING
+
+		//add the items
+		if(!load_items(user, list(tool)))
+			return ITEM_INTERACT_BLOCKING
+		to_chat(user, span_notice("[tool] was added to [src]"))
+		return ITEM_INTERACT_SUCCESS
+
+	//ask player to drag stuff into grinder
+	else if(tool.atom_storage)
+		to_chat(user, span_warning("You must drag & dump contents of [tool] into [src]"))
+		return ITEM_INTERACT_BLOCKING
+
+	return ..()
 
 /obj/machinery/reagentgrinder/wrench_act(mob/living/user, obj/item/tool)
-	. = ..()
-	default_unfasten_wrench(user, tool)
-	return ITEM_INTERACT_SUCCESS
+	if(user.combat_mode)
+		return NONE
 
-/obj/machinery/reagentgrinder/screwdriver_act(mob/living/user, obj/item/tool)
-	. = ITEM_INTERACT_SUCCESS
-	if(!beaker && !length(holdingitems))
-		return default_deconstruction_screwdriver(user, icon_state, icon_state, tool)
+	. = ITEM_INTERACT_BLOCKING
+	if(operating)
+		balloon_alert(user, "still operating")
+		return
 
-/obj/machinery/reagentgrinder/crowbar_act(mob/living/user, obj/item/tool)
-	return default_deconstruction_crowbar(tool)
+	if(default_unfasten_wrench(user, tool) == SUCCESSFUL_UNFASTEN)
```

Comment:
Doesn't this mean that if you start unwrenching, then step away before it finishes, you'll hit it from a tile away?

Reply:
No it returns `ITEM_INTERACTION_BLOCKING` so no hitting. see the above return values